services-camel/src/main/java/edu/unc/lib/cdr/ReplicationProcessor.java

```diff
@@ -0,0 +1,154 @@
+		return replicationLocations.split(";");
+	}
+	
+	private boolean checkReplicationLocations(String[] replicationPaths) {
+		for (String replicationPath : replicationPaths) {
+			if (!Files.exists(Paths.get(replicationPath))) {
+				throw new ReplicationDestinationUnavailableException("Unable to find replication destination {}", replicationPath);
+			}
+		}
+		
+		return true;
+	}
+	
+	private String createFilePath(String basePath, String originalFileChecksum) {
+		String[] binaryFcrepoChecksumSplit = originalFileChecksum.split(":");
+		String checksum = binaryFcrepoChecksumSplit[2];
+		
+		String[] tokens = Iterables.toArray
+				(Splitter.fixedLength(2).split(checksum), 
+						String.class);
+		
+		String remotePath = new StringJoiner("/")
+			.add(basePath)
+			.add(tokens[0])
+			.add(tokens[1])
+			.add(tokens[2])
+			.add(checksum)
+			.toString();
+		
+		return remotePath;
+	}
+	
+	private String createRemoteSubDirectory(String baseDirectory, String binaryChecksum) {
+		String replicationPath = createFilePath(baseDirectory, binaryChecksum);
+		
+		if (!Files.exists(Paths.get(replicationPath))) {
+			new File(replicationPath).mkdirs();
+		}
+
+		return replicationPath;
+	}
+	
+	private String getFileChecksum(String filePath) {
+		String checksum = null;
+		try {
+			checksum = DigestUtils.sha1Hex(new FileInputStream(filePath));
+		} catch (FileNotFoundException e) {
+			log.error("Unable to compute checksum for {}", filePath);
+		} catch (IOException e) {
+			log.error("Unable to compute checksum for {}", filePath);
+		}
+		
+		return checksum;
+	}
+	
+	private String getFilename(String binaryPath) {
+		String[] parts = binaryPath.split("/");
+		return parts[parts.length - 1];
+	}
+	
+	private boolean verifyChecksums(String originalFileChecksum, String replicatedFilePath) {
+		String remoteChecksum = getFileChecksum(replicatedFilePath);
+			
+		if (originalFileChecksum.equals(remoteChecksum)) {
+			log.info("Local and remote checksums match for {}", replicatedFilePath);
+			return true;
+		} else {
+			throw new ReplicationException("Local and remote checksums did not match {} {}", originalFileChecksum, remoteChecksum);
+		}
+	}
+	
+	private void replicate(String binaryPath, String originalFileChecksum, String[] replicationLocations) {
+		try {
+			for (String location : replicationLocations) {
+				String fullPath = Paths.get(createRemoteSubDirectory(location, originalFileChecksum), getFilename(binaryPath)).toString();
+				String[] cmd = new String[]{"rsync", "--update", "--whole-file", "--times", "--verbose", binaryPath, fullPath};
+				Process returnCode = Runtime.getRuntime().exec(cmd);
+				
+				if (returnCode.exitValue() != 0) {
+					throw new ReplicationException("Error replicating {} to {}", binaryPath, fullPath);
+				}
+				
+				verifyChecksums(originalFileChecksum, fullPath);
+			}
+		} catch (IOException e) {
+			log.error("Unable to find file to replicate {}", e);
```

Comment:
Rethrow the exception wrapped as a ReplicationException.  Also, I don't think the exception being caught here necessarily means that the file can't be found.